src/main/java/usi/si/seart/service/GitRepoTopicsService.java

```diff
@@ -0,0 +1,62 @@
+package usi.si.seart.service;
+
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.experimental.FieldDefaults;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+import usi.si.seart.model.GitRepo;
+import usi.si.seart.model.GitRepoTopic;
+import usi.si.seart.model.Topic;
+import usi.si.seart.repository.GitRepoTopicRepository;
+import usi.si.seart.repository.TopicRepository;
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+import javax.persistence.PersistenceContext;
+import javax.validation.constraints.NotNull;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public interface GitRepoTopicsService {
+
+    List<String> getAllTopicsSortByPopularity();
+
+    Topic getOrCreateTopic(@NotNull String languageName);
+
+    void createOrUpdateGitRepoTopics(@NotNull GitRepo repo, @NotNull List<GitRepoTopic> topics);
+
+    @Service
+    @FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)
+    @AllArgsConstructor(onConstructor_ = @Autowired)
+    @PersistenceContext
```

Comment:
Why do we have this annotation here? I know you based this implementation on the analytics service, but here we have no `EntityManager` or associated factory, so I think it's not needed.